src/transforms/detect_exceptions/mod.rs

```diff
@@ -0,0 +1,382 @@
+mod exception_detector;
+mod rules;
+use chrono::Utc;
+pub use exception_detector::*;
+
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+
+use crate::{
+    config::{DataType, Input, Output, TransformConfig, TransformContext, TransformDescription},
+    event::{discriminant::Discriminant, Event},
+    schema,
+    transforms::{TaskTransform, Transform},
+};
+use async_stream::stream;
+use futures::{stream, Stream, StreamExt};
+use std::{collections::HashMap, pin::Pin, time::Duration};
+
+/// ProgrammingLanguages
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq, Hash, Copy)]
+pub enum ProgrammingLanguages {
+    /// Java
+    Java,
+
+    /// Javascript
+    Javascript,
+    Js,
+
+    /// Csharp
+    Csharp,
+
+    /// Python
+    Python,
+    Py,
+
+    // /// Php
+    // Php,
+    /// Go
+    Go,
+
+    /// Ruby
+    Ruby,
+    Rb,
+
+    /// Dart
+    Dart,
+
+    /// All languages
+    All,
+}
+
+/// Configuration for the `detect_exceptions` transform.
+#[serde_as]
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields, default)]
+pub struct DetectExceptionsConfig {
+    /// Programming Languages for which to detect Exceptions
+    ///
+    /// Supported languages are
+    ///   - Java
+    ///   - Python
+    ///   - Go
+    ///   - Ruby
+    ///   - Php
+    ///   - Dart
+    ///   - All (includes all above)
+    #[serde(default = "default_programming_languages")]
+    pub languages: Vec<ProgrammingLanguages>,
+
+    /// The maximum period of time to wait after the last event is received, in milliseconds, before
+    /// a combined event should be considered complete.
+    #[serde(default = "default_expire_after_ms")]
+    #[serde_as(as = "serde_with::DurationMilliSeconds<u64>")]
+    pub expire_after_ms: Duration,
```

Comment:
I'm wondering if `expire_after_ms` and other time intervals should rather be typed `u64` and not Durations. They are used is as integers and we end up with
```
self.expire_after.as_millis().try_into().unwrap()
```
every time we use them, which seems unnecessary.

Reply:
This is how it is used in other transforms, i have simply copied from there.